tools/jinja2-filters.py

```diff
@@ -18,3 +21,19 @@ def filter_b64decode(encoded_value, return_type = "string"):
         return b64decode(encoded_value, validate = True)
     else:
         raise UnsupportedDataType("The encoded value is an unsupported data type")
+
+def filter_chained_get(dictionary_name, chain):
+    if not isinstance(dictionary_name, dict):
+        raise NotADict("The variable does not a dict")
+    if not isinstance(chain, str):
+        raise NotAString("The chain segment does not a string")
+
+    chain_segments = chain.split('.')
+    for chain_segment in chain_segments:
+        if chain_segment in dictionary_name.keys():
+            dictionary_name = dictionary_name[chain_segment]
+            if dictionary_name is None:
+                return ("{{" + chain + "}}")
```

Comment:
Instead of overwriting `dictionary_name` (which is a parameter) please create a local variable. You can call it `value`.